apps/eth_blockchain/test/eth_blockchain/token_test.exs

```diff
@@ -16,7 +16,107 @@ defmodule EthBlockchain.TokenTest do
   use EthBlockchain.EthBlockchainCase, async: true
 
   alias EthBlockchain.Token
-  alias Utils.Helpers.Crypto
+  alias Utils.Helpers.{Crypto, Encoding}
+  alias Keychain.Wallet
+  alias ABI.{TypeEncoder, FunctionSelector}
+
+  setup state do
+    {:ok, {address, _public_key}} = Wallet.generate()
+
+    Map.put(state, :valid_sender, address)
+  end
+
+  describe "deploy_erc20/3" do
+    test "deploys a non mintable contract successfuly when given locked `true`", state do
+      expected_contract_uuid = "3681491a-e8d0-4219-a40a-53d9a47fe64a"
+
+      expected_contract_binary =
+        :eth_blockchain
```

Comment:
If this is extract, as suggested above, the new function could be used here instead of duplicating the code.